app/controllers/org_admin/templates_controller.rb

```diff
@@ -66,31 +66,20 @@ def edit
 
       @current = Template.current(@template.dmptemplate_id)
 
-      if @template == @current
-        # If the template is published
-        if @template.published?
-          # We need to create a new, editable version
-          new_version = Template.deep_copy(@template)
-          new_version.version = (@template.version + 1)
-          new_version.published = false
-          new_version.save
-          @template = new_version
-  #        @current = Template.current(@template.dmptemplate_id)
-        end
-      else
+      if @template != @current
         flash[:notice] = _('You are viewing a historical version of this template. You will not be able to make changes.')
       end
-
       # If the template is published
       if @template.published?
         # We need to create a new, editable version
         new_version = Template.deep_copy(@template)
         new_version.version = (@template.version + 1)
         new_version.published = false
         new_version.save
-        @template = new_version
+        # In order to reflect the new template.id in the URL
+        redirect_to(action: 'edit', id: new_version.id)
```

Comment:
No. Removing this logic prevents the user from being able to view a historical version of a template (go to the history page for a template and select 'View' on an old version). The `flash[:notice]` message warns them that they are viewing a historical copy and unable to edit, but you have redirected them to the current version where they can edit.